aspnetcore/grpc/authn-and-authz.md

```diff
@@ -103,70 +102,76 @@ private static GrpcChannel CreateAuthenticatedChannel(string address)
+        }
+        return Task.CompletedTask;
     });
 ```
 
-A gRPC interceptor can also be used to configure a bearer token. An advantage to using an interceptor is the client factory can be configured to create a new interceptor for each client. This allows an interceptor to be [constructed from DI using scoped and transient services](/dotnet/core/extensions/dependency-injection#service-lifetimes).
+Dependency injection (DI) can be combined with `AddCallCredentials`. An overload passes `IServiceProvider` to the delegate which can be used to get a service [constructed from DI using scoped and transient services](/dotnet/core/extensions/dependency-injection#service-lifetimes).
 
 Consider an app that has:
+
 * A user-defined `ITokenProvider` for getting a bearer token. `ITokenProvider` is registered in DI with a scoped lifetime.
 * gRPC client factory is configured to create clients that are injected into gRPC services and Web API controllers.
 * gRPC calls should use `ITokenProvider` to get a bearer token.
 
 ```csharp
-public class AuthInterceptor : Interceptor
+public interface ITokenProvider
 {
-    private readonly ITokenProvider _tokenProvider;
-    
-    public AuthInterceptor(ITokenProvider tokenProvider)
-    {
-        _tokenProvider = tokenProvider;
-    }
-    
-    public override AsyncUnaryCall<TResponse> AsyncUnaryCall<TRequest, TResponse>(
-        TRequest request,
-        ClientInterceptorContext<TRequest, TResponse> context,
-        AsyncUnaryCallContinuation<TRequest, TResponse> continuation)
+    Task<string> GetTokenAsync();
+}
+
+public class AppTokenProvider : ITokenProvider
+{
+    private string _token;
+
+    public async Task<string> GetTokenAsync()
     {
-        context.Options.Metadata.Add("Authorization", $"Bearer {_tokenProvider.GetToken()}");
-        return continuation(request, context);
+        if (_token == null)
+        {
+            // App code to resolve the token here.
+        }
+
+        return _token;
     }
 }
 ```
 
 ```csharp
+builder.Services.AddScoped<ITokenProvider, AppTokenProvider>();
+
 builder.Services
     .AddGrpcClient<Greeter.GreeterClient>(o =>
     {
         o.Address = new Uri("https://localhost:5001");
     })
-    .AddInterceptor<AuthInterceptor>(InterceptorScope.Client);
+    .AddCallCredentials(async (context, metadata, serviceProvider) =>
+    {
+        var provider = serviceProvider.GetRequiredService<ITokenProvider>();
+        var token = await provider.GetTokenAsync();
+        metadata.Add("Authorization", $"Bearer {token}");
+    }));
 ```
 
-The preceeding code:
-* Defines `AuthInterceptor` which is constructed using the user defined `ITokenProvider`.
+The preceding code:
+
+* Defines `ITokenProvider` and `AppTokenProvider`. These types handle resolving the authentication token for gRPC calls.
+* Registers the `AppTokenProvider` type with DI in a scoped lifetime. `AppTokenProvider` caches the token so only the first call in the scope will need to calculate it.
```

Comment:
Avoiding personification leans to the use of 'require' over 'need.' If 'so that' isn't used, there's an independent clause, so I'll suggest a 'so that' here ...

```suggestion
* Registers the `AppTokenProvider` type with DI in a scoped lifetime. `AppTokenProvider` caches the token so that only the first call in the scope is required to calculate it.
```